unified_planning/engines/mixins/configurator.py

```diff
@@ -0,0 +1,87 @@
+# Copyright 2022 AIPlan4EU project
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+from warnings import warn
+import unified_planning as up
+from typing import Any, Dict, Type
+
+
+class ConfiguratorMixin:
+    """Base class that must be extended by an :class:`~unified_planning.engines.Engine` that is also a `Configurator`."""
+
+    def __init__(self, EngineClass: Type):
+        if not issubclass(EngineClass, up.engines.engine.Engine):
+            raise up.exceptions.UPUsageError(
+                "The given EngineClass is not a subclass of up.engines.Engine!"
+            )
+        self._engine_class = EngineClass
+
+    @staticmethod
+    def is_configurator() -> bool:
+        return True
+
+    @staticmethod
+    def supports_operation_mode_configuration(
```

Comment:
```suggestion
    def supports_operation_mode_for_configuration(
```

but then you shoudl also rename "supports_operation_mode"